rust-src/key_derivation/src/lib.rs

```diff
@@ -34,6 +33,18 @@ impl Net {
     }
 }
 
+impl FromStr for Net {
+    type Err = String;
```

Comment:
Could you please use a proper error type. Using string is not very ergonomic since it does not implement Error.

I suggest
```rust
#[derive(thiserror::Error)]
#[error("{value} is not a valid value for Net")]
pub struct NetFromStrError{value: String}
```